classes/Module/ModuleRepository.php

```diff
@@ -0,0 +1,152 @@
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0  Academic Free License (AFL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\AutoUpgrade\Module;
+
+use PrestaShop\Module\AutoUpgrade\AutoupgradeException;
+
+/**
+ * Class ModuleRepository is used to get lists of modules (those on the disk, native modules by version, ...)
+ */
+class ModuleRepository
+{
+    /** @var string */
+    private $modulesDir;
+
+    /** @var string */
+    private $disabledModulesDir;
+
+    /** @var AddonsClientInterface */
+    private $addonsClient;
+
+    /** @var array */
+    private $modulesByVersion;
+
+    /**
+     * @param string $modulesDir
+     * @param string $disabledModulesDir
+     * @param AddonsClientInterface $addonsClient
+     */
+    public function __construct(
+        $modulesDir,
+        $disabledModulesDir,
+        AddonsClientInterface $addonsClient
+    ) {
+        $this->modulesDir = rtrim($modulesDir, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR;
+        $this->disabledModulesDir = rtrim($disabledModulesDir, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR;
+        $this->addonsClient = $addonsClient;
+        $this->modulesByVersion = array();
+    }
+
+    /**
+     * Returns the list of modules present in the modules folder, the array returned
+     * contains the technical names of the modules.
+     *
+     * @return array
+     */
+    public function getModulesOnDisk()
+    {
+        return $this->listModulesFromDirectory($this->modulesDir);
+    }
+
+    /**
+     * Returns the list of modules present in the disabled modules folder, the array returned
+     * contains the technical names of the modules.
+     *
+     * @return array
+     */
+    public function getDisabledModulesOnDisk()
+    {
+        return $this->listModulesFromDirectory($this->disabledModulesDir);
+    }
+
+    /**
+     * Returns the list of native modules for the specified version, the array returned
+     * contains the technical names of modules indexed by their Addons id.
+     *
+     * @param string $version
+     *
+     * @return array|false
+     */
+    public function getNativeModulesForVersion($version)
+    {
+        if (empty($this->modulesByVersion[$version])) {
+            $this->modulesByVersion[$version] = $this->addonsClient->request('native', ['version' => $version]);
+        }
+
+        return $this->modulesByVersion[$version];
+    }
+
+    /**
+     * Return the list of custom modules on disk (a custom module is any module that is
+     * not native). The array returned contains the technical names of the modules.
+     *
+     * @param array $versions
+     *
+     * @return array
+     */
+    public function getCustomModulesOnDisk($versions)
+    {
+        $modulesOnDisk = $this->getModulesOnDisk();
+        $nativeModules = array();
```

Comment:
Don't mix short array and `array`

Reply:
Right! I'm used to short array now, but in this module I think use the `array()` notation since we're not sure of the PHP version? wdyt?

Reply:
Short array came with 5.4 and this module require 5.6+.